backend/msgpack/server.go

```diff
@@ -182,12 +194,45 @@ func (s *server) write(
 		s.RUnlock()
 		return err
 	}
-	numShards := placement.NumShards()
-	shard := s.shardFn(id, numShards)
-	instances := placement.InstancesForShard(shard)
-	err = s.writerMgr.WriteTo(instances, shard, mu, pl)
+	var (
+		numShards = placement.NumShards()
+		shardID   = s.shardFn(id, numShards)
+		instances = placement.InstancesForShard(shardID)
```

Comment:
The agg tier placement tooling will ensure that the cutover time of a new placement is set such that all clients will receive the placement before the actual shard cutover happens at the agg tier. I'll add you to an email thread where the difference between a placement cutover and a shard cutover is discussed in more detail.